onnxscript/rewriter/pattern_base_test.py

```diff
@@ -106,6 +106,50 @@ def always_fail_condition(context, x):
         # Should return None due to failing condition
         self.assertIsNone(match_result)
 
+    def test_pattern_impl_no_match_returns_match_object(self):
+        """Test that PatternImpl.match returns match object (not always None) when available."""
+        
+        def identity_pattern(op, x):
+            return op.Identity(x)
+        
+        pattern_impl = pattern.PatternImpl(identity_pattern, name="IdentityPattern")
+        
+        # Create a model with an Add node (should not match Identity pattern)
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17]>
+            agraph (float[N] x, float[N] y) => (float[N] z)
+            {
+                z = Add(x, y)
+            }
+        """
+        )
+        model = ir.serde.deserialize_model(model_proto)
+        
+        # Find the Add node
+        add_node = None
+        for node in model.graph:
+            if node.op_type == "Add":
+                add_node = node
+                break
+        
+        self.assertIsNotNone(add_node)
+        
+        # Test pattern matching - should fail because Add != Identity
+        match_result = pattern_impl.match(model, model.graph, add_node)
+        
+        # The result should either be None (GenericPatternMatcher) or a failed MatchResult (SimplePatternMatcher)
```

Comment:
Can you simplify this test to assert that `bool(match_result)` is false? We can separately cleanup the differences between the two different matchers. I think it would be better to have them uniformly return a MatchResult in future.